repository/github_user.go

```diff
@@ -0,0 +1,96 @@
+//Package repository container the fuctions to handle githubusers model information
+package repository
+
+import (
+	"encoding/csv"
+	"io"
+	"log"
+	"lolidelgado/github-users/models"
+	"os"
+	"strconv"
+)
+
+type GithubUser struct {
+	githubUserCsvFileName string
+}
+
+type IGithubUserRepository interface {
+	FetchAll() ([]models.GithubUser, error)
+}
+
+const staticFilesPath = "static/"
+
+func NewGithubUser(fileName string) *GithubUser {
+	return &GithubUser{
+		fileName,
+	}
+}
+
+func (g *GithubUser) FetchAll() ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	return arrayToGithubUser(lines), nil
+}
+
+func (g *GithubUser) GetById(id int) ([]models.GithubUser, error) {
+	lines, err := g.readCsv()
+	if err != nil {
+		return nil, err
+	}
+	var user [][]string
+	for _, line := range lines {
+		if idFromFile, _ := strconv.Atoi(line[0]); idFromFile == id {
+			user = append(user, line)
+			break
+		}
+	}
+	if len(user) > 0 {
+		return arrayToGithubUser(user), nil
+	} else {
+		return []models.GithubUser{}, nil
+	}
```

Comment:
Pro-tip, you can get rid off the else statement as following:

```suggestion
	if len(user) > 0 {
		return arrayToGithubUser(user), nil
	}
	
	return []models.GithubUser{}, nil
```